graylog2-server/src/test/java/org/junit/jupiter/engine/descriptor/ContainerMatrixTestsDescriptor.java

```diff
@@ -87,10 +94,24 @@ public ContainerMatrixTestsDescriptor(TestDescriptor parent,
         this.mongoVersion = MongodbServer.DEFAULT_VERSION;
         this.extraPorts.addAll(extraPorts);
         this.mongoDBFixtures.addAll(mongoDBFixtures);
+        this.withMailServerEnabled = false;
     }
 
-    protected static String createKey(Lifecycle lifecycle, String mavenProjectDirProvider, String pluginJarsProvider, SearchVersion esVersion, MongodbServer mongoVersion) {
-        return "Lifecycle: " + lifecycle.name() + ", MavenProjectDirProvider: " + mavenProjectDirProvider + ", PluginJarsProvider: " + pluginJarsProvider + ", Search: " + esVersion + ", MongoDB: " + mongoVersion.getVersion();
+    protected static String createKey(Lifecycle lifecycle, String mavenProjectDirProvider, String pluginJarsProvider, SearchVersion searchVersion, MongodbServer mongoVersion, boolean withMailServerEnabled) {
+        final List<Pair<String, Object>> values = Lists.newArrayList(
```

Comment:
You could also use an `ImmutableMap.Builder` here and iterate over the entry set later. This would be a more common idiom and does not require us to depend on Apache Commons.

Reply:
Good point, done, thanks!